pattern_generator.rb

```diff
@@ -0,0 +1,79 @@
+class PatternGenerator
+  
+  PATTERNELEMENTS = {"."=>["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"], "#"=>['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']}
+
+  def verify(input, pattern)
+    verify = true
+    pattern.split('').zip(input.split('')).each do |compare|
+      if compare[0] == compare[1] || PATTERNELEMENTS[compare[0]].include?(compare[1])
+        next
+      else
+        verify = false 
+        break       
+      end
+    end
+    verify
+  end
+
+  def total_available(pattern)
+    avail_array = pattern.split('').map do |item|
+      if item == "." 
+        item = 26
+      elsif item == "#"
+        item = 10
+      else
+        item = nil
+      end
+    end
+    avail_array.compact.reverse.reduce(&:*)
+   end
+
+  def create_enumerated_array(pattern)
+    @pattern_arr = pattern.split('')
+
+    @pattern_arr.count(".").times do 
```

Comment:
instance variables are nice, but try to make an initialize and create `attr_readers` or `attr_accessors` when needed